tests/operator_test.go

```diff
@@ -780,6 +781,119 @@ var _ = Describe("ALL Operator tests", func() {
 				}, 2*time.Minute, 1*time.Second).Should(BeTrue())
 			})
 		})
+
+		var _ = Describe("Priority class tests", func() {
+			var (
+				cdi                   *cdiv1.CDI
+				systemClusterCritical = cdiv1.CDIPriorityClass("system-cluster-critical")
+				osUserCrit            = &schedulev1.PriorityClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "openshift-user-critical",
+					},
+					Value: 10000,
+				}
+			)
+			f := framework.NewFramework("operator-priority-class-test")
+			verifyPodPriorityClass := func(prefix, priorityClassName, labelSelector string) {
+				Eventually(func() string {
+					controllerPod, err := utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, prefix, labelSelector)
+					if err != nil {
+						return ""
+					}
+					return controllerPod.Spec.PriorityClassName
+				}, 2*time.Minute, 1*time.Second).Should(BeEquivalentTo(priorityClassName))
+			}
+
+			BeforeEach(func() {
+				cr, err := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), "cdi", metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					Skip("CDI CR 'cdi' does not exist.  Probably managed by another operator so skipping.")
+				}
+				Expect(err).ToNot(HaveOccurred())
+				cdi = cr
+				if cr.Spec.PriorityClass != nil {
+					By(fmt.Sprintf("Current priority class is: [%s]", *cr.Spec.PriorityClass))
+				}
+			})
+
+			AfterEach(func() {
+				if cdi == nil {
+					return
+				}
+
+				cr, err := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), "cdi", metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				cr.Spec.PriorityClass = cdi.Spec.PriorityClass
+
+				_, err = f.CdiClient.CdiV1beta1().CDIs().Update(context.TODO(), cr, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				if !utils.IsOpenshift(f.K8sClient) {
+					Eventually(func() bool {
+						return errors.IsNotFound(f.K8sClient.SchedulingV1().PriorityClasses().Delete(context.TODO(), osUserCrit.Name, metav1.DeleteOptions{}))
+					}, 2*time.Minute, 1*time.Second).Should(BeTrue())
+				}
+				By("Ensuring the CDI priority class is restored")
+				prioClass := ""
+				if cr.Spec.PriorityClass != nil {
+					prioClass = string(*cr.Spec.PriorityClass)
+				}
+				// Deployment
+				verifyPodPriorityClass(cdiDeploymentPodPrefix, string(prioClass), common.CDILabelSelector)
+				// API server
+				verifyPodPriorityClass(cdiApiServerPodPrefix, string(prioClass), "")
+				// Upload server
+				verifyPodPriorityClass(cdiUploadProxyPodPrefix, string(prioClass), "")
+				By("Verifying there is just a single cdi controller pod")
+				Eventually(func() error {
+					_, err := utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, cdiDeploymentPodPrefix, common.CDILabelSelector)
+					return err
+				}, 2*time.Minute, 1*time.Second).Should(BeNil())
+				By("Ensuring this pod is the leader")
```

Comment:
why we need this check?

Reply:
Because we just changed the cdi deployment pod, and if we just happen to run a leader election test next, it will fail because the leader election hasn't happened yet. So this ensures that the deployment pod is a single pod, AND it is the current active leader. So any following tests won't be messed up.